b2sdk/bucket.py

```diff
@@ -750,6 +750,34 @@ def hide_file(self, file_name):
         response = self.api.session.hide_file(self.id_, file_name)
         return FileVersionInfoFactory.from_api_response(response)
 
+    def copy_file(
+        self,
+        file_id,
+        new_file_name,
+        bytes_range=None,
+        metadata_directive=None,
+        content_type=None,
+        file_info=None,
+    ):
+        """
+        Creates a new file by copying from an existing file.
+
+        :param str file_id: file ID of existing file
+        :param str new_file_name: file name of the new file
+        :param tuple[int, int], None bytes_range: start and end offsets
+        :param str, None metadata_directive: `'COPY'` or `'REPLACE'`, default is `'COPY'`
+        :param str, None content_type: content_type for the new file if `'REPLACE'`, default will copy the content_type of old file
+        :param dict, None file_info: file_info for the new file if `'REPLACE'`, default will copy the file_info of old file
+        """
+        return self.api.session.copy_file(
+            file_id,
```

Comment:
We should be passing the bucket object here. Current server restricts the copy to same bucket, but we should not rely on that.

Reply:
but the API doesn't take the bucket argument.

Reply:
🤦‍♂ I'll discuss on slack

Reply:
Use `destinationBucketId` parameter (it is not documented and doesn't really work yet, so please make a comment that as of 2019-06-24 we are implementing this for forward-compatibility of the client library. Add a `..note:` that says it is up to the server to decide what is allowed and what is not, state for today being that only intra-bucket copies are permitted)